data/lib/repositories/dishes_repository_impl.dart

```diff
@@ -1,14 +1,36 @@
+import 'package:core/core.dart';
+import 'package:data/entities/dishes/entities/dish_entity.dart';
+import 'package:data/mappers/mappers.dart';
 import 'package:data/providers/firebase_provider.dart';
+import 'package:data/providers/hive_provider.dart';
 import 'package:domain/domain.dart' as domain;
+import 'package:domain/domain.dart';
 
 class DishesRepositoryImpl implements domain.DishesRepository {
   final FirebaseProvider _firebaseProvider;
+  final HiveProvider _hiveProvider;
+  final MapperFactory mapper;
 
-  DishesRepositoryImpl({required FirebaseProvider firebaseProvider})
-      : _firebaseProvider = firebaseProvider;
+  DishesRepositoryImpl({
+    required this.mapper,
+    required FirebaseProvider firebaseProvider,
+    required HiveProvider hiveProvider,
+  })  : _firebaseProvider = firebaseProvider,
+        _hiveProvider = hiveProvider;
 
   @override
   Future<List<domain.Dish>> fetchDishes(domain.NoParams payload) async {
-    return await _firebaseProvider.fetchDishes();
+    final List<Dish> dishes;
+    if (await InternetConnection.isInternetConnection()) {
+      dishes = await _firebaseProvider.fetchDishes();
```

Comment:
create variable like this
 final bool hasInternetConnection =
        await InternetConnectionInfo.checkInternetConnection();
and then write:
    if (hasInternetConnection) {
don't use 'await' inside statement